database.go

```diff
@@ -63,6 +64,29 @@ func sameNameOnDB(newBook Book) (unique bool) {
 	}
 }
 
+var bookNotFound error
+
+/* Search a book in database based on ID and returns it if succeed. */
+func searchById(id uuid.UUID) (Book, error) {
+	sqlStatement := `SELECT id, name, price, inventory FROM bookstable WHERE id=$1;`
+	foundRow := dbObject.QueryRow(sqlStatement, id)
+	var bookToReturn Book
+	err := foundRow.Scan(&bookToReturn.ID, &bookToReturn.Name, &bookToReturn.Price, &bookToReturn.Inventory)
+	if err != nil {
+		switch err {
+		case sql.ErrNoRows:
+			{
+				bookNotFound = sql.ErrNoRows //Does it resolve the problem of using SQL on main() side or it's just an useless step?
```

Comment:
It does not, use the `errResponse` for now.
Create a new one like `ErrBookNotFound` in the `errors.go` file.

Reply:
I've tried to use errResponse in many ways before, but it is not of error type, that is what I have to return. Furthermore, I thougth that if is not good pratice to use SQL on main() side, in the same way it is not responsibility of database.go handle with http responses. Then I looked for something in the middle of the two options.

Reply:
Fair. Create an error. Pretty simple:
```go
var ErrBookNotFound = errors.New("book not found")
```

Simple, return that. Than test it in the main execution with `errors.Is()`.